pybricksdev/compile.py

```diff
@@ -123,12 +124,26 @@ async def compile_multi_file(path: str, abi: int):
     parts: List[bytes] = []
 
     for name, module in finder.modules.items():
+        if not module.__file__:
+            continue  # system module
         mpy = await compile_file(module.__file__, abi)
 
         parts.append(len(mpy).to_bytes(4, "little"))
         parts.append(name.encode() + b"\x00")
         parts.append(mpy)
 
+    # look for .mpy modules
+    for name in finder.any_missing():
+        for path in sys.path:
+            try:
+                with open(os.path.join(path, f"{name}.mpy"), "rb") as f:
+                    mpy = f.read()
```

Comment:
We could also validate if the mpy version is compatible here and fail early rather than waiting for a runtime failure after it is already sent to the hub.

Reply:
Is there an easy way to parse the version?

Reply:
https://docs.micropython.org/en/latest/reference/mpyfiles.html#the-header

The `mpy[0]` should be `b"M"` and `mpy[1]` should be the version number which should match the `abi` argument.